### PR TITLE
Performance - contributors component refactor

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

Immediately display the contributors modal button to reduce LCP.

Other changes:
- Used the MdChevronRight icon to match the initial close state and avoid the initial animation causing the LCP to be higher sometimes
- Removed the Skeletons and add one inside the modal

====
### Tests after changes
Webpagetest https://www.webpagetest.org/result/240430_BiDc8E_92E/

All the LCPs are the same now, ~2/3s.

### CODE REVIEW

File: src/components/SideNavMobile.tsx [hunk 1]
```diff
@@ -108,11 +108,19 @@ const NavLink = ({ item, path, toggle }: NavLinkProps) => {
...
             animate={isOpen ? "open" : "closed"}
           >
-            <Icon as={MdExpandMore} boxSize={6} color="secondary" />
+            <Icon
+              as={MdExpandMore}
+              // display by default the icon in this position `>` (close state)
+              // this will prevent the icon from rotating when the page loads
+              transform="rotate(-90deg)"
+              boxSize={6}
+              color="secondary"
```
Comment (minimalsm) [inline src/components/SideNavMobile.tsx:122]: Should we just use `MdChevronRight` where rightward facing arrows are default state?

File: src/components/SideNav.tsx [hunk 2]
```diff
@@ -13,13 +13,13 @@ import docLinks from "../data/developer-docs-links.yaml"
 
 export const dropdownIconContainerVariant = {
   open: {
-    rotate: 0,
+    rotate: 90,
     y: 3,
```
Comment (minimalsm) [inline src/components/SideNav.tsx:17]:
```suggestion
```

The framer positioning can be removed due to the flex container being added.

File: src/components/SideNav.tsx [hunk 3]
```diff
@@ -13,13 +13,13 @@ import docLinks from "../data/developer-docs-links.yaml"
...
     transition: {
       duration: 0.4,
     },
   },
-  closed: { rotate: -90, y: 0 },
+  closed: { rotate: 0, y: 0 },
```
Comment (minimalsm) [inline src/components/SideNav.tsx:22]:
```suggestion
  closed: { rotate: 0, },
```

File: src/components/SideNavMobile.tsx (hunk 1 above)
Comment (pettinarip) [reply src/components/SideNavMobile.tsx:122]: Aha! makes total sense, yup. Didn't know we had that icon.

File: src/components/SideNav.tsx (hunk 2 above)
Comment (pettinarip) [reply src/components/SideNav.tsx:17]: Ah right! nice catch. Lets remove that as well.

**Review by minimalsm — APPROVED** [review]

**Comment by nloureiro** [comment]
yes, this is the temporary change we agreed on the other day.

Can I ask just to add the same spacing above and below the title?
I think it's `var(--eth-space-8)` between the title and the button

Screen Shot 2024-04-30 03 28 54 PM

**Comment by pettinarip** [comment]
@nloureiro fixed

**Review by nloureiro — APPROVED** [review]

**Review by nhsz — APPROVED** [review]

**Pull request merged**